flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/mathExpressions.scala

```diff
@@ -243,6 +243,19 @@ case class Atan(child: Expression) extends UnaryExpression {
   }
 }
 
+case class Atan2(left: Expression, right: Expression) extends BinaryExpression with InputTypeSpec {
```

Comment:
We should not use `InputTypeSpec` here. It forces a certain type which results in BigDecimal even for double types.